manifests/hourly.pp

```diff
@@ -19,13 +19,8 @@
 ) {
   assert_private()
 
-  $dir_ensure = $logrotate::ensure_cron_hourly ? {
-    'absent'  => $logrotate::ensure_cron_hourly,
-    'present' => 'directory'
-  }
-
   file { "${logrotate::rules_configdir}/hourly":
-    ensure => $dir_ensure,
+    ensure => 'directory',
```

Comment:
Why not allow this to be removed when hourly is ensure=absent?

Reply:
When which parameter is set not /set ?

* `$logrotate::manage_cron_hourly` controls if the crons are managed.
* $logrotate::manage_systemd_timer controls if the units should be managed

Just noticed this confusion in the class:

In `logrotate::rule` it includes logrotate::daily if and only if there is a rule which is `hourly`.

However there is also a `contain` in logrotate::init  - that contain seems wrong to me. Should we be creating these units, crons and directories if there are no hourly rules.

In reference to the `"${logrotate::rules_configdir}/hourly":` I don't see how you can know up front that the directory need to be created or not - certainly not that it can be deleted.

I would remove that contain from manifests::init myself so for the default case the extra cron, timer and directory never get created.

Reply:
Looking at this ancient MR https://github.com/voxpupuli/puppet-logrotate/pull/174/files it looks to be quite deliberate
that the `${logrotate::rules_configdir}/hourly"` is always created in all circumstances.

Either that should go or the one here in `logrotate::rule`

```
case $rotate_every {
 'hour', 'hourly': {
   include logrotate::hourly
   $rule_path = "${logrotate::rules_configdir}/hourly/${rulename}"
```

one of these is definitely redundant.

Reply:
Could split hourly in two files.
Move the directory management into its own file to create and purge always.
Existing file without that could be included iff there an hourly logrotate rule.